Launcher.py

```diff
@@ -170,30 +173,18 @@ class Launcher(App):
         base_title: str = "Archipelago Launcher"
         container: ContainerLayout
         grid: GridLayout
-
-        _tools = {c.display_name: c for c in components if c.type == Type.TOOL}
-        _clients = {c.display_name: c for c in components if c.type == Type.CLIENT}
-        _adjusters = {c.display_name: c for c in components if c.type == Type.ADJUSTER}
-        _miscs = {c.display_name: c for c in components if c.type == Type.MISC}
+        _tool_layout: ScrollBox
+        _client_layout: ScrollBox
 
         def __init__(self, ctx=None):
             self.title = self.base_title
             self.ctx = ctx
             self.icon = r"data/icon.png"
+            self._tool_layout = ScrollBox()
+            self._client_layout = ScrollBox()
```

Comment:
I think those would typically be in `build()` instead of `__init__()`. It's a bit awkward with members being undefined after init, but I think that's how we expect kivy to work.

Reply:
Yeah, I don't like members not being assigned at the end of `__init__`, so I made them `Optional[ScrollBox]` so we can create them in build.